httpx/_multipart.py

```diff
@@ -78,23 +78,44 @@ def __init__(self, name: str, value: FileTypes) -> None:
 
         fileobj: FileContent
 
+        headers: typing.Dict[str, str] = {}
+        content_type: typing.Optional[str] = None
+
+        # This large tuple based API largely mirror's requests' API
+        # It would be good to think of better APIs for this that we could include in httpx 2.0
+        # since variable length tuples (especially of 4 elements) are quite unwieldly
         if isinstance(value, tuple):
             try:
-                filename, fileobj, content_type = value  # type: ignore
+                filename, fileobj, content_type, headers = value  # type: ignore
             except ValueError:
-                filename, fileobj = value  # type: ignore
-                content_type = guess_content_type(filename)
+                # 4th parameter (headers) not included
+                try:
+                    filename, fileobj, content_type = value  # type: ignore
+                except ValueError:
+                    # neither the 3rd parameter (content_type) nor the 4th (headers) was included
+                    filename, fileobj = value  # type: ignore
```

Comment:
An alternative would be...

```python
# Unpack filename, fileobj, and optionally content_type and headers.
filename, fileobj, content_type, headers, *_ = value + (None, None)
```

Not sure if the more concise version is better or worse here.

Or we could do explicitly length-based...

```python
if len(value) == 2:
    filename, fileobj = value
elif len(value) == 3:
    filename, fileobj, content_type = value
else:
    filename, fileobj, content_type, headers = value
```

Looking at it now, I think the length based version is probably neatest, since it's just very clear and obvious. Marginally nicer than the various indents that try..catch requires.

Reply:
Yep yep. I guess I just took what was there and extended it. Obviously it doesn't scale, I like matching on the length more.

Reply:
Ok changed it to a length based comparison